src/string_pool.cr

```diff
@@ -178,6 +178,6 @@ class StringPool
     hasher = Crystal::Hasher.new
     hasher = str.to_slice(len).hash(hasher)
     # hash should be non-zero, so `or` it with high bit
-    hasher.result | 0x8000000000000000_u64
+    hasher.result.to_u64 | 0x8000_0000_0000_0000_u64
```

Comment:
You should use 32/64 bit arithmetic here or simply use 64 bit hashes everywhere

Reply:
This is exactly what I propose in the updated PR description.

I chose the easy solution here, but tweaking StringPool to take whatever is easy (except for high not or).

Reply:
I'm strongly vote for `UInt64` type. Just because I have unfinished work that hardly depends on it (number normalization).